falguard.py

```diff
@@ -69,13 +69,28 @@ def __init__(self, spec_path):
     def __call__(self, request, response, resource, params):
         self.process_resource(request, response, resource, params)
 
+    def _build_http_not_allowed_error(self, uri):
+        # Error 405 requires we provide the list of allowed methods
+        # for this URI. If None is found, then we produce a 404 error.
+        allowed_methods = [method for method in falcon.HTTP_METHODS
+                           if self._spec.get_op_for_request(method, uri)]
```

Comment:
This way get_op_for_request will run twice for the original method.

Reply:
Well, I was wondering, which is cheaper:
- one more call to `get_op_for_request` (that does a dict key lookup inside) ;
- as many comparisons as they are elements in `HTTP_METHODS`?